aiida/orm/nodes/data/msonable.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+"""Data plugin for classes that implement the ``MSONable`` class of the ``monty`` library."""
+import importlib
+
+from monty.json import MSONable
+
+from aiida.orm import Data
+
+
+class MsonableData(Data):
+    """Data plugin that allows to easily wrap objects that are MSONable.
+
+    To use this class, simply construct it passing an isntance of any ``MSONable`` class and store it, for example:
+
+        from pymatgen.core import Molecule
+
+        molecule = Molecule(['H']. [0, 0, 0])
+        node = MsonableData(molecule)
+        node.store()
+
+    After storing, the node can be loaded like any other node and the original MSONable instance can be retrieved:
+
+        loaded = load_node(node.pk)
+        molecule = loaded.obj
+
+    .. note:: As the ``MSONable`` mixin class requires, the wrapped object needs to implement the methods ``as_dict``
+        and ``from_dict``. A default implementation should be present on the ``MSONable`` base class itself, but it
+        might need to be overridden in a specific implementation.
+
+    """
+
+    def __init__(self, obj, *args, **kwargs):
```

Comment:
Add type checking

Reply:
I added type checking, but since we are very flexible, all we can say is `typing.Any` pretty much. Or is there some better typing we can have still?

Reply:
The best way is to define a Protocol: https://docs.python.org/3/library/typing.html#typing.Protocol, although you need the typing_extensions backport package for python < 3.8

Reply:
(See e.g. https://github.com/executablebooks/markdown-it-py/blob/1d27af1da41fd3b00da8307040e6e274925437b2/markdown_it/renderer.py#L21)